### PR TITLE
Make updates to fix tests, and make fewer breaking changes for existing users

### PR DESCRIPTION
**Pull request by kkamperschroer**
We don't want to make unecessary breaking changes to existing methods. Re-introduce the second parameter `token` for `onRegistered` and `onUnregistered`.

Also made some fixes to existing tests that we managed to run locally.

Pairs @brannon @flegallo @mpodwysocki

### CODE REVIEW

File: sdk/build.gradle [hunk 1]
```diff
@@ -7,8 +7,8 @@ buildscript {
     }
     dependencies {
         classpath 'com.android.tools.build:gradle:3.3.2'
-        classpath 'com.google.gms:google-services:3.0.0'
-        
+        classpath 'com.google.gms:google-services:4.0.1'
```
Comment (dhei) [inline sdk/build.gradle]: Thanks @kkamperschroer for bumping to latest.

File: sdk/src/notifications-handler/src/main/java/com/microsoft/windowsazure/notifications/NotificationsManager.java [hunk 2]
```diff
@@ -35,34 +35,26 @@
...
-                            }
-
-                        }
-                    });
+                setHandler(context, notificationsHandlerClass);
 
+                FirebaseInstanceId.getInstance().getInstanceId().addOnSuccessListener(new OnSuccessListener<InstanceIdResult>() {
+                    @Override
+                    public void onSuccess(InstanceIdResult instanceIdResult) {
+                    NotificationsHandler handler = getHandler(context);
```
Comment (dhei) [inline sdk/src/notifications-handler/src/main/java/com/microsoft/windowsazure/notifications/NotificationsManager.java]: nit: Fix indentation.

File: sdk/src/notifications-handler/src/main/java/com/microsoft/windowsazure/notifications/NotificationsManager.java [hunk 3]
```diff
@@ -35,34 +35,26 @@
...
 
-                } catch (Exception e) {
```
Comment (dhei) [inline sdk/src/notifications-handler/src/main/java/com/microsoft/windowsazure/notifications/NotificationsManager.java]: I am not sure if this try...catch was to handle any specific error cases or redundant "just-in-case" code. I don't see any obvious reasons for requiring it, though it doesn't hurt.

Comment (brannon) [reply sdk/src/notifications-handler/src/main/java/com/microsoft/windowsazure/notifications/NotificationsManager.java]: Removing it didn't generate any warnings. None of the code we are calling declares that it throws.

Comment (dhei) [reply sdk/src/notifications-handler/src/main/java/com/microsoft/windowsazure/notifications/NotificationsManager.java]: Then let's remove it.

File: sdk/src/notifications-handler/src/main/java/com/microsoft/windowsazure/notifications/NotificationsManager.java (hunk 2 above)
Comment (dhei) [reply sdk/src/notifications-handler/src/main/java/com/microsoft/windowsazure/notifications/NotificationsManager.java]: will address in another PR.

File: sdk/src/notifications-handler/src/main/java/com/microsoft/windowsazure/notifications/NotificationsManager.java (hunk 3 above)
Comment (dhei) [reply sdk/src/notifications-handler/src/main/java/com/microsoft/windowsazure/notifications/NotificationsManager.java]: will address it in another PR.

**Pull request merged**